cornell-dti/course-plan
# Add entrance/grad seasons to onboarding

### Summary 

This pull request adds entrance and grad season inputs to the Onboarding modal. For new users, they default to Fall & Spring, and for existing users, they will not be able to re-submit onboarding until they are filled out.

This solves the issue in #618 where students that enter in Spring 2020 would be given the post Fall 2020 A&S reqs since we don't check semester season, just year. This also now implements @vaishnavi17 's code in #529 to base starter semesters off of entrance and grad seasons instead of just years

- [x] Added season dropdowns in onboarding with emojis. Initial choice is dependent on if user is new, see above.
- [x] Updated the review page to show the dropdown info
- [x] Updated error check code to block users from submitting unless seasons selected
- [x] Update Cypress tests to select semester seasons

### Test Plan 

1. Test that for existing users, opening onboarding has 'Select one' placeholder text in the new dropdowns, and cannot be submitted until both are filled out. 
![image](https://user-images.githubusercontent.com/25535093/154336739-45ca9aca-c3a2-4287-bcc2-4e163d9b9b6b.png)

2. Test editing the fields and re-opening modal has season selections saved, and review page is accurate
![image](https://user-images.githubusercontent.com/25535093/154336973-cee7efab-7d5d-47ed-b1f8-c01de705b4e1.png)

3. Delete `user-onboarding-data` document for myself and confirm the the new flow, and Fall and Spring respectively selected by default
![image](https://user-images.githubusercontent.com/25535093/154336801-3a4684c5-2397-477a-ac79-dc7d39991b89.png)

4. Confirm #529 works - specifically that a user's starter semesters populate based on their starting and graduating season now (delete `user-onboarding-data` and `user-semesters)
 
![image](https://user-images.githubusercontent.com/25535093/154337932-14b6b17a-1b0b-4b09-bb6c-05fb3e5e5fe1.png)
(when selecting Spring 2018 and Fall 2022 as entrance/grad)

### Notes 

#618 can be modified after this is merged to use a user's entrance semester!

Some future suggestions based on this PR to make items for (are any of these a priority?)
1. Update TrackUsers analytics to change grad/entrance year counters to grad/entrance sem counters to we know how many spring admits etc. there are
2. Update Vaish's code to handle entrance or grad semesters besides Fall and Winter - selecting Spring or Winter does not populate the correct semesters
